app/memorize.py

```diff
@@ -0,0 +1,136 @@
+from functools import wraps
+import hashlib
+import pickle
+
+from wrapt import ObjectProxy
+
+
+class CacheMissError(RuntimeError):
+    """
+    Exception thrown when memorized object is not found in cache and
+    error_on_miss is set to True.
+    """
+    pass
+
+
+class MemorizedObject(ObjectProxy):
+    """
+    Thin wrapper around any memorized objects, adding attributes indicating
+    the object's cache key and whether or not it was returned from cache.
+    Mostly used to make logging and testing easier.
+    """
+    def __init__(self, wrapped):
+        super(MemorizedObject, self).__init__(wrapped)
+        self._self_from_cache = False
+        self._self_cache_key = None
+
+    def __reduce__(self):
+        """
+        Makes MemorizedObject instances picklable, if the wrapped object is
+        picklable itself.
+        """
+        return type(self), (self.__wrapped__,)
+
+    def __repr__(self):
+        """
+        When inspecting MemorizedObject instances in a REPL, we want to show
+        the wrapped object, rather than <MemorizedObject ...>
+        """
+        return repr(self.__wrapped__)
+
+    @property
+    def from_cache(self):
+        return self._self_from_cache
+
+    @from_cache.setter
+    def from_cache(self, value):
+        self._self_from_cache = value
+
+    @property
+    def cache_key(self):
+        return self._self_cache_key
+
+    @cache_key.setter
+    def cache_key(self, value):
+        self._self_cache_key = value
+
+
+class memorize(object):
+    """
+    Method decorator to memoize that method in memcached based on the name
+    of the method, the name of the class to whom that method belongs, and the
+    call signature.
+
+    Example:
+
+    class Book(object):
+        def __init__(self, title, author):
+            self.title = title
+            self.author = author
+
+        @memorize(cxn)
```

Comment:
It seems like every `@memorize` invocation passes in the same memcached pointer. Maybe set that as the default, so that memorize.py fully abstracts memcached?